src/librustc_typeck/diagnostics.rs

```diff
@@ -4690,7 +4646,6 @@ register_diagnostics! {
     E0224, // at least one non-builtin train is required for an object type
     E0227, // ambiguous lifetime bound, explicit lifetime bound required
     E0228, // explicit lifetime bound required
-    E0231, // only named substitution parameters are allowed
```

Comment:
Don't these need to be commented out instead of deleted? I don't know.

Reply:
Moved to `librustc/diagnostics`, not deleted.